clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -83,24 +85,35 @@ func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.Meas
 	if err != nil {
 		return summaries, err
 	}
+	var wg wait.Group
+	errList := util.NewErrorList()
+	var lock sync.Mutex
 	runningRuntimeObjects := 0
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(defaultWaitForRCPodsInterval) {
-		if runningRuntimeObjects >= len(runtimeObjectsList) {
-			return summaries, nil
-		}
-
-		if err := waitForRuntimeObject(config.ClientSet, runtimeObjectsList[runningRuntimeObjects], timeout-time.Since(start)); err != nil {
-			return summaries, fmt.Errorf("waiting for %v error: %v", kind, err)
-		}
-		runningRuntimeObjects++
-		if namespace == metav1.NamespaceAll {
-			glog.Infof("%s: %d / %d %ss with all pods running", waitForControlledPodsRunningName, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		} else {
-			glog.Infof("%s: %v: %d / %d %ss with all pods running", waitForControlledPodsRunningName, namespace, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		}
-	}
-
-	return summaries, fmt.Errorf("timeout while waiting for %ss to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running with all pods", kind, namespace, labelSelector, fieldSelector, runningRuntimeObjects)
+	for objectCouter := 0; objectCouter < len(runtimeObjectsList); objectCouter++ {
+		objectIndex := objectCouter
+		wg.Start(func() {
```

Comment:
@shyamjvs Assuming that we agree on approach for parallelism in #218, I will add the same solution here.